starlarktime/module.go

```diff
@@ -0,0 +1,155 @@
+//
+//   time = module(
+//      parse,
+//      now,
+//      time,
+//   )
+//
+// def parse(string, format[, location]):
+//   The parse function parses the given "string" with the "format" specified.
+//   Formats follow the specifications in the Golang "time" package.
+//   The optional "location" argument can be used to specify the timezone
+//   location for the resulting time. In case no location is given and the
+//   given string does not contain any timezone information, the time is
+//   returned in UTC.
+//
+//   For example
+//     parse("2020-01-01T12:00:00Z", "2006-01-02T15:04:05Z", "Europe/Berlin")
+//   will return a time object with the given time localized in the CET/CEST
+//   timezone.
+//
+// def now([location]):
+//   The now function returns the current time.
+//   The optional "location" parameter can be used to specify the timezone
+//   location for the returned time. In case no location is given, the local
+//   time is returned.
+//
+// def time([year[, month[, day[, hour[, minute[, second[, nanosecond[, location]]]]]]]]):
+//   The time function returns the a new time instance with the given values.
+//   All parameters are optional and default to zero if not specified. The "location"
+//   parameter can be used to specify the timezone location for the returned
+//   time. It defaults to UTC if not specified.
+//
+var Module = &starlarkstruct.Module{
+	Name: "time",
+	Members: starlark.StringDict{
+		"parse": starlark.NewBuiltin("time.parse", parse),
+		"now": starlark.NewBuiltin("time.now", now),
+		"time": starlark.NewBuiltin("time.time", new),
+	},
+}
+
+func parse(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var value, fmt, tz starlark.String
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 2, &value, &fmt, &tz); err != nil {
+		return nil, err
+	}
+
+	var t time.Time
+	var err error
+	if len(tz) < 1 {
+		t, err = time.Parse(string(fmt), string(value))
+		if err != nil {
+			return starlark.None, err
+		}
+	} else {
+		loc, err := time.LoadLocation(string(tz))
+		if err != nil {
+			return starlark.None, err
+		}
+		t, err = time.ParseInLocation(string(fmt), string(value), loc)
+		if err != nil {
+			return starlark.None, err
+		}
+	}
+
+	return &StarlarkTime{Time: t}, nil
+}
+
+func now(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var tz starlark.String
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 0, &tz); err != nil {
+		return nil, err
+	}
+
+	t := time.Now()
+	if len(string(tz)) > 0 {
+		loc, err := time.LoadLocation(string(tz))
+		if err != nil {
+			return starlark.None, err
+		}
+		t = t.In(loc)
+	}
+
+	return &StarlarkTime{Time: t}, nil
+}
+
+func new(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+ y := starlark.MakeInt(0)
```

Comment:
BTW: I would accept a PR to UnpackPositionalArgs to make it accept {int,uint}{,8,16,32,64}, not just plain old int, and do appropriate range checks. Then all of this could be:

var y, m, d, h, min, s, ns int

(You already don't need the AsInt32 checks below, because the type of &y etc tells Unpack to demand an int argument value.)